### ISSUE TITLE
Building a specific blog workflow with Nikola

### ISSUE THREAD

**Issue by benjaminweb** [getnikola/nikola#2689]
This is *no* bug report, however an issue that IMHO deserves its own place here.

____

I want to check whether I can configure/build a blogging tool based on Nikola's core components to meet the following specific requirements:

I want to write articles in a single file:

```
2017-03-06

This is the text of the first article.
As it has only three lines,
it will not have a link pointing to its full content.

2017-03-05

This is the second article.
It has many lines.
Which will necessitate a link
that points to the full article content.
The index will display only
the first five lines.
```

Any ``YYYY-MM-DD`` will result in a new article.

Full syntax would be: ``YYYY-MM-DD [OPTIONAL TITLE] #tag1 #tag2``

An article does not require a title at all. However, a date must be provided.

Two conditions:
 * article lasts five or less lines: no link for the full article.
 * article lasts more than five lines: a link references to the full article.

Both are listed on the index page with the first five lines.

A **calendar** highlights dates with at least one article.
In addition, a different index page is rendered: 
```
YEAR | MONTH      01        02        03        04        05        06        07        08        09
2017                       * * * * 
2018…
```

Each asterisk is a link to an article. Here, five articles have been posted in February, 2017.
Unlike common indexes that know only a single vector for time (scrolling up and down), this calendar index has one vertical (years) and one horizontal (monthly). It should increase the number of items displayed significantly by not showing the title. Date and title will be shown, instead as a tooltip. This is intended to shorten the time span from all articles to the wanted considerably.

Can I implement such a specific solution with Nikola? In addition I would be interested in at least one non-Nikola alternative for reference. If, how could I implement the solution with Nikola?

**Comment by ralsina** [comment]
The calendar sounds doable with some templating work. Maybe a small change
in the calendar plugin.
I don't understand the description of the  article format.
As for the five lines thing, currently that is manual but you could hack it
to be auto.

This sounds a lot like a job description, so if you like I can offer you my
standard consulting fee.

No idea how other alternatives would work, look around.

**Comment by benjaminweb** [comment]
@ralsina Thanks for your timely comment. I've updated the issue: It includes a full example of the format. Sorry for not being precise straight out. Is that now clearer to you, @ralsina?

Thanks for offering me your professional services. I would prefer to implement it on my own.
As such, I haven't concluded completely if Nikola is the best thing for what I want.

**Comment by Kwpolska** [comment]
What you want to implement is troublesome, and I don't think most other
SSGs would allow to have more than one post in a file. The problem is, it's
hard to parse this reasonably. You see, your filesystem knows where every
file is on the hard drive. With your format, if it was meant to be human
friendly, the software would need to search for the dates on every build,
which can take some time. Moreover, a lot of places in Nikola assume that 1
input file = 1 post.

Why not let your filesystem do its job and write posts in separate files?

(Your calendar idea is generally reasonable and doable; the 5 lines thing
might require some templating magic or changing behavior in core)

-- 
Chris Warrick <https://chriswarrick.com/>

**Comment by ralsina** [comment]
Well, it could be preprocessed into a sane format, assuming @benjaminweb can write something to split the data into input files for Nikola. 

As part of that preprocessing it can manually add the teaser comments to "long" posts.

So: for the input requirement, it's either a small script or a Nikola plugin.

**Comment by Kwpolska** [comment]
(The preprocessing cannot be reasonably done as part of the Nikola build process)

**Comment by Kwpolska** [comment]
Closing due to no activity.
